elastalert/elastalert.py

```diff
@@ -534,43 +540,57 @@ def get_hits_terms(self, rule, starttime, endtime, index, key, qk=None, size=Non
+        query = rule['filter'][0]['query_string']['query']
+        data, count = self.get_ch_data(rule, starttime, endtime, agg_key, query)
+        
+        if data is None:
+            return {}
+        payload = {rule['metric_agg_key']+"_"+rule['metric_agg_type']: {'value': data}}
+
+        self.num_hits += count
+        return {endtime: payload}
+
+    def get_error_rate(self, rule, starttime, endtime):
+        agg_key = rule['metric_agg_type']+"()"
+        query = rule['filter'][0]['query_string']['query']
+        total_data, total_count = self.get_ch_data(rule, starttime, endtime, agg_key, query)
+
+        if total_data is None:
+            return {}
+        
+        query = query+" AND "+rule['error_condition']
+        error_data, error_count = self.get_ch_data(rule, starttime, endtime, agg_key, query)
+
+        if error_data is None:
+            return {}
+
+        payload = {'error_count': error_data, 'total_count': total_data}
+        self.num_hits += int(total_count)
+
+        return {endtime: payload}
+
+    def get_ch_data(self, rule, starttime, endtime, agg_key, freshquery):
+        data = {
+                    "selects":[],
+                    "start_time":dt_to_ts(starttime),
+                    "end_time":dt_to_ts(endtime),
+                    "freshquery": freshquery,
+                    "group_bys":[],
+                    "sort_orders":[{"sort_by": agg_key,"sort_direction":"desc"}],
+                    "limit":"500",
+                    "aggregations":[{"function": rule['metric_agg_type'].upper(), "field": rule['metric_agg_key']}]
+                }
         try:
-            if not rule['five']:
-                res = self.current_es.search(
-                    index=index,
-                    doc_type=rule.get('doc_type'),
-                    body=query,
-                    search_type='count',
-                    ignore_unavailable=True
-                )
-            else:
-                res = self.current_es.search(index=index, doc_type=rule.get('doc_type'), body=query, size=0, ignore_unavailable=True)
-        except ElasticsearchException as e:
+            res = requests.post(self.query_endpoint, json=data)
+            res.raise_for_status()
+        except requests.exceptions.RequestException as e:
             if len(str(e)) > 1024:
                 e = str(e)[:1024] + '... (%d characters removed)' % (len(str(e)) - 1024)
             self.handle_error('Error running query: %s' % (e), {'rule': rule['name']})
-            return None
-        if 'aggregations' not in res:
-            return {}
-        if not rule['five']:
-            payload = res['aggregations']['filtered']
-        else:
-            payload = res['aggregations']
-        self.num_hits += res['hits']['total']
-        return {endtime: payload}
+            return None,0
+        res = json.loads(res.content)
+        return res['data'][0][agg_key], res['rows']
```

Comment:
What would be there in these two values?

Reply:
agg value and the number of rows satisfied the filter criteria